### PR TITLE
Update free-programming-books-langs.md

### PR DESCRIPTION
**Pull request by Khushi-Shukla**
## What does this PR do?
Add resource(s) | Remove resource(s) | Add info | Improve repo
Add resource

### Why is this valuable (or not)?
It is valuable as it provides fundamental concepts about ruby on rails

### How do we know it's really free?
It's free as I have attached the link to pdf 

### For book lists, is it a book? For course lists, is it a course? etc.
It is a book

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction).
- [x] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: books/free-programming-books-langs.md [hunk 1]
```diff
@@ -2229,6 +2229,7 @@ Books on general-purpose programming that don't focus on a specific language are
 * [Just Enough Ruby to Get By](http://jasonkim.ca/projects/just_enough_ruby_to_get_by/)
 * [Learn Ruby First](https://essenceofchaos.gitbooks.io/learn-ruby-first/content/) - Frederick John
 * [Learn Ruby in Y minutes](https://learnxinyminutes.com/docs/ruby/)
+* [Learn Ruby on Rails](https://learn-rails.com/products/learn-rails-1.pdf) - Daniel Kehoe (PDF)
```
Comment (sohansai) [inline books/free-programming-books-langs.md:2232]: Book already in the repository, available at: [link](https://github.com/EbookFoundation/free-programming-books/blob/main/books/free-programming-books-langs.md#ruby-on-rails)

**Comment by Khushi-Shukla** [comment]
Added a different book. Please check.

File: books/free-programming-books-langs.md [hunk 2]
```diff
@@ -2222,6 +2222,7 @@ Books on general-purpose programming that don't focus on a specific language are
 
 * [A community-driven Ruby style guide](https://github.com/bbatsov/ruby-style-guide)
 * [Core Ruby Tools](https://launchschool.com/books/core_ruby_tools) - Launch School (HTML)
+* [Design Patterns in Ruby](https://github.com/abdelrahman-hsn/ruby-ebooks/blob/master/Design%20Patterns%20in%20Ruby%2C%20Addison-Wesley%20(2008).pdf) - Addison-Wesley
```
Comment (sohansai) [inline books/free-programming-books-langs.md:2225]: This is not an authorized distribution. Refer to docs/CONTRIBUTING.md for guidelines.

**Comment by Khushi-Shukla** [comment]
Please review the change. @sohansai

**Pull request merged**